content/cif-connector/src/main/content/jcr_root/libs/commerce/gui/components/authoring/editor/pagepreview/clientlibs/pagepreview.js

```diff
@@ -24,58 +24,55 @@ window.CIF.PagePreview = {};
 
     var relPdpPreview = '.cq-commerce-pdp-preview-activator';
     var relPlpPreview = '.cq-commerce-plp-preview-activator';
+    var productPreviewServletUrl = '/bin/wcm/cif.previewproduct.html';
+    var categoryPreviewServletUrl = '/bin/wcm/cif.previewcategory.html';
 
-    var handlePreview = function(e, data) {
+    var handleProductPreview = function(e, data) {
+        handlePreview(data, productPreviewServletUrl);
+    };
+
+    var handleCategoryPreview = function(e, data) {
+        handlePreview(data, categoryPreviewServletUrl);
+    };
+
+    var handlePreview = function(data, previewServletUrl) {
         if (!data) {
             return;
         }
 
         var selections = data.selections;
-        var productSlug = null;
+        var identifier = null;
         if (Array.isArray(selections)) {
             if (selections.length > 0 && selections[0]) {
-                productSlug = selections[0].value;
+                identifier = selections[0].value;
             }
         } else if (selections) {
-            productSlug = selections.value;
+            identifier = selections.value;
         }
 
-        var url = Granite.author.ContentFrame.location;
-        var previewUrl = createPreviewUrl(url, productSlug);
-
-        if (previewUrl) {
-            window.open(previewUrl);
-        }
-    };
-
-    var createPreviewUrl = function(url, selector) {
-        if (!url || !selector) {
-            return null;
-        }
-
-        var indLastSlash = url.lastIndexOf('/');
-        if (indLastSlash < 0 || indLastSlash === url.length - 1) {
+        // Currently the picker returns only one identifier for the selected item.
+        // It can be one of the following: <id>, <url_key>, <path>, <sku>, <sku>#<variant_sku>
+        // This needs to be fixed in the future version so the picker will return multiple identifiers
+        var [slug, variantSku] = identifier ? identifier.split('#') : [];
+        if (!slug) {
             return null;
         }
 
-        var urlPath = url.slice(0, indLastSlash + 1);
-        var urlName = url.substring(indLastSlash + 1);
-        var indHtml = urlName.lastIndexOf('.html');
-        var previewUrl = null;
-        if (indHtml > -1) {
-            var indSelector = urlName.slice(0, indHtml).lastIndexOf('.');
-            if (indSelector > -1) {
-                previewUrl = urlPath + urlName.slice(0, indSelector) + '.' + selector + '.html';
-            } else {
-                previewUrl = urlPath + urlName.slice(0, indHtml) + '.' + selector + '.html';
-            }
+        // prepare all possible parameters
+        var params = previewServletUrl === productPreviewServletUrl ? { url_key: slug, sku: slug } : { id: slug };
```

Comment:
I think this is also confusing, because that `slug` (which honestly here I don't know if it's really the `sku` or the `url_key`) is used for both ... if we only support one particular use case, let's write the code for what it really does support today. I understand that you might want to support future use cases, but IMO this adds more confusion because I don't really know what works and what not.

Reply:
The original implementation implied only `selectionId=slug` is used for the pickers. That's what we tried to fix here. 
I agree it's not ideal but as long as the picker supports all those identifiers we need to account for them here as well. But in the selection handler we cannot distinguish between the identifiers. Variable names can be changed as suggested in previous comment. 
Also same code exists in addon fix [here](https://git.corp.adobe.com/CIF/cif-on-skyline/blob/96d400b1da7a44bc345da9a466fbecbb4390acb5/apps/src/main/content/jcr_root/libs/commerce/gui/components/authoring/editor/pagepreview/clientlibs/pagepreview.js#L56-L69)